common/lib/protocol-definitions/src/clients.ts

```diff
@@ -5,53 +5,137 @@
 
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode - either view-only ("read") or allowing edits ("write").
+ *
+ * @remarks Note: this should not be confused with user permissions.
+ * If a user does not have write privileges, they will still not be permitted to make edits, even if the
+ * connection mode is set to "write".
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * TODO
+ */
 export interface ICapabilities {
+	/**
+	 * Indicates if the client represents a potentially interactive session with a user (if 'true') or if it's a "system entity" (if 'false').
+	 *
+	 * @remarks
+	 *
+	 * The only "system entity" scenario at the moment is the
+	 * {@link https://fluidframework.com/docs/concepts/summarizer/ | summarizer client}.
+	 */
 	interactive: boolean;
 }
+
+/**
+ * TODO
+ */
 export interface IClientDetails {
+	/**
+	 * {@inheritDoc ICapabilities}
+	 */
 	capabilities: ICapabilities;
+
+	/**
+	 * TODO: What is this? Are there specific expected values? What does it mean for this to be undefined?
+	 */
 	type?: string;
+
 	/**
 	 * If the environment needs to specify multiple properties which gives info about the environment, then
-	 * it should be in particular format like: "prop1:val1;prop2:val2;prop3:val3"
+	 * it should be in particular format like: `prop1:val1;prop2:val2;prop3:val3`.
+	 *
+	 * TODO: What does it mean for this to be undefined?
 	 */
 	environment?: string;
+
+	/**
+	 * TODO: What is this? What does it mean for this to be undefined?
+	 */
 	device?: string;
 }
 
+/**
+ * TODO
+ */
 export interface IClient {
+	/**
+	 * {@inheritDoc ConnectionMode}
+	 */
 	mode: ConnectionMode;
+
+	/**
+	 * {@inheritDoc IClientDetails}
+	 */
 	details: IClientDetails;
+
+	/**
+	 * TODO: What is this? Are there specific values we expect?
+	 */
 	permission: string[];
+
+	/**
+	 * The user associated with this client.
```

Comment:
maybe user -> user information?
and client -> connection?
Note that in some very rare situations user identity can change over multiple connections (I do not remember details), so it's really data that is valid for juts that connection